tests/test_e2e.py

```diff
@@ -0,0 +1,143 @@
+import lightgbm as lgb
+import numpy as np
+import pytest
+
+try:
+    # neptune-client=0.9.0+ package structure
+    from neptune.new import init_run
+    from neptune.new.integrations.lightgbm import (
+        NeptuneCallback,
+        create_booster_summary,
+    )
+except ImportError:
+    # neptune-client>=1.0.0 package structure
+    from neptune import init_run
+    from neptune.integrations.lightgbm import (
+        NeptuneCallback,
+        create_booster_summary,
+    )
+
+
+@pytest.mark.parametrize("log_importances", [True, False])
+@pytest.mark.parametrize("log_confusion_matrix", [True, False])
+@pytest.mark.parametrize("log_pickled_booster", [True, False])
+@pytest.mark.parametrize("log_trees", [True, False])
+@pytest.mark.parametrize("log_trees_as_dataframe", [True, False])
+def test_e2e(dataset, log_importances, log_confusion_matrix, log_pickled_booster, log_trees, log_trees_as_dataframe):
```

Comment:
This generates 2^5 -> 32 tests.

Reply:
How long does it run?

Reply:
It takes 2 to 9 mins on different platforms

<img width="1244" alt="image" src="https://user-images.githubusercontent.com/19503980/218116677-e673d1b8-11ef-475b-8c99-a6609d2242e5.png">

Reply:
They are independent. Will reduce the combination (and I don't think that would change).